subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -43,6 +43,16 @@
 import java.util.function.Supplier;
 
 public interface UnitOfWork extends Describable {
+
+    /**
+     * String that identifies the type of the work.
+     * <p>
+     * This is used, so we don't need to use the class name to distinguish between different types of work.
+     */
+    default String getWorkType() {
+        return getClass().getName();
+    }
```

Comment:
I didn't want to introduce identifiers for every unit of work, especially if we won't capture them right now. I introduced one for `TaskExecution` since I expected us to capture that somewhat "soonish". Though for the current step we need something for transform executions, all the other ones don't matter. So I suppose the right thing to do here is to only introduce the "TRANSFORM" work type and leave the other ones to be the class name of the unit of work. WDYT?

Reply:
I don't like what this might default to. I wouldn't like to end up with some things captured via a shortname like `TASK` and others with a FQCN.

If we want to signal that some units of work are not to be captured this way, maybe return `null` or `Optional.empty()`, WDYT?

Reply:
I do not feel strongly either way, but I also agree that mixing class names with explicit labels like `TRANSFORM` does not seem to be useful. We wouldn't show this string to the users directly, and any other use implies having a contract in place (like possible pre-defined values).

I suggest making this method return a nullable value. For now, only one implementation will be returning `"TRANSFORM"`, and it will be null in all other cases. Between Optional and null I would go for null, because many other methods in the `UnitOfWork` interface use `@Nullable`s.

I would also suggest changing the name to something like `getBuildOperationWorkType()`. I think it makes it clearer why it's nullable/optional. Also, I don't see at the moment how keeping the name generic would be useful in contexts other than the corresponding build operation.

@lptr, would this approach work?

Reply:
👍 for the name. I'm actually in favor of `Optional`. I'd like to convert all null-returning methods to use `Optional` going forward, just never gotten around to it yet...

Reply:
I updated the PR